e2e/blog-test.sh

```diff
@@ -0,0 +1,165 @@
+
+    aws acm-pca delete-certificate-authority --certificate-authority-arn $CA_ARN --permanent-deletion-time-in-days 7
+
+}
+
+clean_up() {
+    set +e
+
+    kubectl delete -f $E2E_DIR/blog-test/test-nlb-tls-app.yaml >/dev/null 2>&1
+
+    kubectl delete -f $E2E_DIR/blog-test/nlb-lab-tls.yaml >/dev/null 2>&1
+
+    kubectl delete -f $E2E_DIR/blog-test/test-cluster-issuer.yaml >/dev/null 2>&1
+
+    helm uninstall aws-load-balancer-controller -n kube-system >/dev/null 2>&1
+
+    delete_ca
+
+}
+
+install_aws_load_balancer() {
+    kubectl create serviceaccount aws-load-balancer-controller -n kube-system >/dev/null 2>&1
+    kubectl annotate serviceaccount aws-load-balancer-controller -n kube-system iam.amazonaws.com/role=$OIDC_IAM_ROLE >/dev/null 2>&1
+    helm repo add eks https://aws.github.io/eks-charts >/dev/null 2>&1
+    kubectl apply -k "github.com/aws/eks-charts/stable/aws-load-balancer-controller//crds?ref=master" >/dev/null 2>&1
+    helm install aws-load-balancer-controller eks/aws-load-balancer-controller -n kube-system --set clusterName=$CLUSTER_NAME --set serviceAccount.create=false --set serviceAccount.name=aws-load-balancer-controller --set image.repository=docker.io/amazon/aws-alb-ingress-controller --set env.AWS_ACCESS_KEY_ID="$AWS_ACCESS_KEY_ID" --set env.AWS_SECRET_ACCESS_KEY="$AWS_SECRET_ACCESS_KEYS" --set env.AWS_REGION="$AWS_REGION" >/dev/null 2>&1
+}
+
+main() {
+
+    set_variables
+
+    DEPLOYMENT_NAME=$(kubectl get deployments -n $K8S_NAMESPACE -ojson | jq -r ".items[0].metadata.name")
+
+    if [ -z "$DEPLOYMENT_NAME" ]; then
+        echo "[ERROR] Found empty ACK controller deployment name. Exiting ..."
+        exit 1
+    fi
+
+    echo "$DEPLOYMENT_NAME deployment found."
+
+    POD_NAME=$(kubectl get pods -n $K8S_NAMESPACE -ojson | jq -r ".items[0].metadata.name")
+
+    if [ -z "$POD_NAME" ]; then
+        echo "[ERROR] Found empty ACK controller pod name. Exiting ..."
+        exit 1
+    fi
+
+    echo "$POD_NAME pod found."
+
+    install_aws_load_balancer
+
+    echo "AWS Load Balancer installed."
+
+    envsubst <$E2E_DIR/blog-test/cluster-issuer.yaml >$E2E_DIR/blog-test/test-cluster-issuer.yaml
+
+    kubectl apply -f $E2E_DIR/blog-test/test-cluster-issuer.yaml 1>/dev/null || exit 1
+
+    kubectl apply -f $E2E_DIR/blog-test/nlb-lab-tls.yaml 1>/dev/null || exit 1
+
+    CERTIFICATE_NAME=$(kubectl get certificate -ojson | jq -r ".items[0].metadata.name")
+
+    if [ -z "$CERTIFICATE_NAME" ]; then
+        echo "[ERROR] Found empty certificate name. Exiting ..."
+        exit 1
+    fi
+
+    echo "$CERTIFICATE_NAME certificate found."
+
+    envsubst <$E2E_DIR/blog-test/nlb-tls-app.yaml >$E2E_DIR/blog-test/test-nlb-tls-app.yaml
+
+    kubectl create serviceaccount aws-load-balancer-controller >/dev/null 2>&1
+    kubectl annotate serviceaccount aws-load-balancer-controller iam.amazonaws.com/role=$OIDC_IAM_ROLE >/dev/null 2>&1
```

Comment:
Same as the comment above, I don't really think the IRSA is being used here so I think we can just safely remove this.

Reply:
Removed serviceaccount